census_view.cpp

```diff
@@ -932,13 +936,839 @@ inline std::vector<std::string> const& CensusViewDataViewModel::all_headers() co
+            );
+
+        wxGridCellDateEditor::BeginEdit(row, col, grid);
+    }
+};
+
+class DatumSequenceEditor
+    :public wxGridCellEditor
+{
+  public:
+    DatumSequenceEditor() = default;
+
+    // We don't define a copy ctor because wxGridCellEditor doesn't provide one
+    // and its derived classes are supposed to override Clone() to support
+    // polymorphic copying, instead of using the copy ctor.
+    DatumSequenceEditor(DatumSequenceEditor const&) = delete;
+    DatumSequenceEditor& operator=(DatumSequenceEditor const&) = delete;
+
+    void Create(wxWindow* parent, wxWindowID id, wxEvtHandler* evtHandler) override;
+
+    void BeginEdit(int row, int col, wxGrid* grid) override;
+    bool EndEdit
+        (int row
+        ,int col
+        ,wxGrid const* grid
+        ,wxString const& oldval
+        ,wxString* newval
+        ) override;
+    void ApplyEdit(int row, int col, wxGrid* grid) override;
+
+    void Reset() override;
+
+    wxGridCellEditor* Clone() const override;
+
+    wxString GetValue() const override;
+
+  private:
+    InputSequenceEntry* Entry() const;
+
+    wxString     value_;
+    Input const* input_ {nullptr};
+    std::string  field_;
+};
+
+void DatumSequenceEditor::Create
+    (wxWindow*     parent
+    ,wxWindowID    id
+    ,wxEvtHandler* evtHandler
+    )
+{
+    m_control = new(wx) InputSequenceEntry(parent, id, "sequence_editor");
+
+    wxGridCellEditor::Create(parent, id, evtHandler);
+}
+
+void DatumSequenceEditor::BeginEdit(int row, int col, wxGrid* grid)
+{
+    auto table = grid->GetTable();
+    value_ = table->GetValue(row, col);
+    input_ = &row_at(*table, row);
+    field_ = col_name(*table, col);
+
+    auto entry = Entry();
+
+    entry->text_ctrl().SetValue(value_);
+    entry->input(*input_);
+    entry->field_name(field_);
+}
+
+bool DatumSequenceEditor::EndEdit
+    (int
+    ,int
+    ,wxGrid const*
+    ,wxString const&
+    ,wxString*       newval
+    )
+{
+    auto value = Entry()->text_ctrl().GetValue();
+
+    if(value == value_)
+        {
+        return false;
+        }
+
+    value_ = value;
+
+    *newval = GetValue();
+
+    return true;
+}
+
+void DatumSequenceEditor::ApplyEdit(int row, int col, wxGrid* grid)
+{
+    grid->GetTable()->SetValue(row, col, value_);
+}
+
+void DatumSequenceEditor::Reset()
+{
+    Entry()->text_ctrl().SetValue(value_);
+}
+
+wxGridCellEditor* DatumSequenceEditor::Clone() const
+{
+    auto editor = new(wx) DatumSequenceEditor();
+    editor->value_ = value_;
+    editor->input_ = input_;
+    editor->field_ = field_;
+
+    return editor;
+}
+
+wxString DatumSequenceEditor::GetValue() const
+{
+    return Entry()->text_ctrl().GetValue();
+}
+
+InputSequenceEntry* DatumSequenceEditor::Entry() const
+{
+    auto entry = dynamic_cast<InputSequenceEntry*>(m_control);
+    LMI_ASSERT(entry);
+    return entry;
+}
+
+// This class is used to implement conversion to and from cells string for use
```

Comment:
I think this would be more clear:
```suggestion
// This class is used to implement conversion to and from strings shown in the grid cells
```